Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -627,7 +628,15 @@ Task<bool> SwitchContentAsync(Page view, bool animated, bool removed = false, bo
 					fragments.Add(fragment);
 				}
 			}
-			transaction.Commit();
+
+			if (Element.On<Xamarin.Forms.PlatformConfiguration.Android>().IsAllowingStateLoss())
+			{
+				transaction.CommitAllowingStateLoss();
+			}
+			else
+			{
+				transaction.Commit();
+			}
```

Comment:
You should respond to property changes. See https://github.com/xamarin/Xamarin.Forms/pull/407/files#L98